pkg/controller/deployment/sync.go

```diff
@@ -145,7 +145,7 @@ func (dc *DeploymentController) getNewReplicaSet(d *apps.Deployment, rsList, old
 		rsCopy := existingNewRS.DeepCopy()
 
 		// Set existing new replica set's annotation
-		annotationsUpdated := deploymentutil.SetNewReplicaSetAnnotations(d, rsCopy, newRevision, true)
+		annotationsUpdated := deploymentutil.SetNewReplicaSetAnnotations(d, rsCopy, newRevision, true, 262144)
```

Comment:
The number 262144 in my error message was measured in characters, but here it looks like you're passing it as the max number of elements in the revision history list. That said, this is a huge amount of revision history, I think we should cap this at something low-ish and arbitrary so that we don't end up carrying around a massive annotation (like 100 maybe?).

Reply:
This limit is for number of characters.
Since we don't know per element length, using element count is inaccurate.

Reply:
> Since we don't know per element length, using element count is inaccurate.

Yeah, I recognize that it'd be inaccurate to count elements instead of characters, but the point I was trying to make is that I don't think we should be letting the length of the annotation reach anywhere near the max allowed annotation length since that is a whopping 256KB which bloats the memory in all the master processes and etcd and all the API responses. If I had just 20 ReplicaSets with 256KB annotations, that would already be 5MB. 

The 64 bit max int represented as a string uses 19 characters and iff we add a comma, that's 20 characters. Keeping the cap on this annotation at 262144 characters means that if every element in the list consumed 20 characters, the history would be able to hold 13,107 elements! Can you think of a reason why someone might need such a large amount of revision history?

If we capped the list at 100 elements then the max length it could possibly be is only 2000 characters so we'd never need to worry about hitting the annotation size limit. If you think 100 isn't high enough, then capping it at 500 elements would still only be 10,000 characters and 1000 would only be 20,000 characters.

Reply:
I'd say 100 element (~2000 chars) is probably enough. Please change it to 2000 and make it a `const`.